sys/net/gnrc/network_layer/icmpv6/echo/gnrc_icmpv6_echo.c

```diff
@@ -107,4 +107,121 @@ void gnrc_icmpv6_echo_req_handle(gnrc_netif_t *netif, ipv6_hdr_t *ipv6_hdr,
     }
 }
 
+static void _fill_payload(uint8_t *buf, size_t len, uint32_t now)
+{
+    uint8_t i = 0;
+
+    if (len >= sizeof(uint32_t)) {
+        memcpy(buf, &now, sizeof(now));
+        len -= sizeof(now);
+        buf += sizeof(now);
+    }
+
+    while (len--) {
+        *buf++ = i++;
+    }
+}
+
+static void _check_payload(const void *buf, size_t len, uint32_t now,
+                           uint32_t *triptime, int *corrupt)
+{
+    uint8_t i = 0;
+    const uint8_t *data = buf;
+
+    if (len >= sizeof(uint32_t)) {
+        *triptime = now - unaligned_get_u32(buf);
+        len  -= sizeof(uint32_t);
+        data += sizeof(uint32_t);
+    }
+
+    while (len--) {
+        if (*data++ != i++) {
+            *corrupt = data - (uint8_t *)buf - 1;
+            break;
+        }
+    }
+}
+
+int gnrc_icmpv6_echo_send(const gnrc_netif_t *netif, const ipv6_addr_t *addr,
+                          uint16_t id, uint16_t seq, uint8_t ttl, size_t len)
+{
+    int res = 0;
+    gnrc_pktsnip_t *pkt, *tmp;
+    ipv6_hdr_t *ipv6;
+    uint8_t *databuf;
+
+    pkt = gnrc_icmpv6_echo_build(ICMPV6_ECHO_REQ, id, seq, NULL, len);
+    if (pkt == NULL) {
+        DEBUG("error: packet buffer full\n");
+        return -ENOMEM;
+    }
+
+    databuf = (uint8_t *)(pkt->data) + sizeof(icmpv6_echo_t);
+    tmp = gnrc_ipv6_hdr_build(pkt, NULL, addr);
+    if (tmp == NULL) {
+        puts("error: packet buffer full");
+        goto error_exit;
+    }
+    pkt = tmp;
+    ipv6 = pkt->data;
+    /* if ttl is unset (i.e. 0) gnrc_ipv6 will select hop limit */
+    ipv6->hl = ttl;
+    if (netif != NULL) {
+        tmp = gnrc_netif_hdr_build(NULL, 0, NULL, 0);
+        if (tmp == NULL) {
+            DEBUG("error: packet buffer full");
+            res = -ENOMEM;
+            goto error_exit;
+        }
+        gnrc_netif_hdr_set_netif(tmp->data, netif);
+        pkt = gnrc_pkt_prepend(pkt, tmp);
+    }
+
+    /* add TX timestamp & test data */
+    _fill_payload(databuf, len, ztimer_now(ZTIMER_USEC));
```

Comment:
Any specific reason why you do not call `ztimer_now()` within `_fill_payload()` and changed the signiture of that function compared to `master`?

Reply:
Just for symetry with `_check_payload()`